src/fits_schema/binary_table.py

```diff
@@ -56,56 +59,54 @@ class BinaryTableHeader(HeaderSchema):
     EXTNAME = HeaderCard(required=False, type_=str)
 
 
-class Column(metaclass=ABCMeta):
-    """A binary table column descriptor.
-
-    Attributes
-    ----------
-    unit: astropy.units.Unit
-        unit of the column
-    strict_unit: bool
-        If True, the unit must match exactly, not only be convertible.
-    required: bool
-        If this column is required (True) or optional (False)
-    name: str
-        Use to specify a different column name than the class attribute name.
-    ndim: int
-        Dimensionality of a single row, numbers have ndim=0.
-        The resulting data column has ``ndim_col = ndim + 1``
-    shape: Tuple[int]
-        Shape of a single row.
-    """
+@dataclass
+class Column(SchemaElement, metaclass=ABCMeta):
+    """A binary table column descriptor."""
+
+    #: allow compatible units if False
+    strict_unit: bool = False
+
+    #: dimensionality of column
+    ndim: int | None = None
+
+    #: specify exact shape (length of each dimension)
+    shape: tuple[int] | None = None
+
+    # the data type of the column
+    dtype: type = None
 
-    def __init__(
-        self,
-        *,
-        unit=None,
-        strict_unit=False,
-        required=True,
-        name=None,
-        ndim=None,
-        shape=None,
-        description: str = "",
-    ):
-        self.required = required
-        self.unit = unit
-        self.strict_unit = strict_unit
-        self.name = name
-        self.shape = shape
-        self.ndim = ndim
-        self.description = description
+    def __post_init__(self):
+        """Check the schema."""
+        super().__post_init__()
 
         if self.shape is not None:
-            self.shape = tuple(shape)
+            self.shape = tuple(self.shape)
             # Dimensionality of the table is one more than that of a single row
             if self.ndim is None:
                 self.ndim = len(self.shape)
             elif self.ndim != len(self.shape):
-                raise ValueError(f"Shape={shape} and ndim={ndim} do not match")
+                raise ValueError(
+                    f"Shape={self.shape} and ndim={self.ndim} do not match"
+                )
         else:
             # simple column by default
             if self.ndim is None:
                 self.ndim = 0
+        if self.name:
+            self._check_name()
+
+    def _check_name(self):
+        """Ensure column name follows FITS conventions.
+
+        The FITS standard recommends only upper and lower-case letters and
```

Comment:
Recommends. So I think this should be a warning, not an error. You can then use war rings configuration to turn specific warnings into errors.

Reply:
good idea. Done.